### ISSUE TITLE
Significant Discrepancies Between Experimental Results and Paper

### ISSUE THREAD

**Issue by LiZhi-CASIA** [CVLAB-Unibo/crossmodal-feature-mapping#6]
Hello! Thank you for sharing this excellent work with the community. I’ve encountered some challenges in reproducing the results reported in the paper; specifically, my experimental outcomes show considerable differences, particularly for certain categories. I’ve taken care to align my setup with the environment and procedures described in the paper, yet the results still diverge noticeably. I would appreciate any guidance or insights you could offer.

Experimental Setup:
GPU: NVIDIA GeForce RTX 3090
Dataset used: MVTec 3D-AD
Preprocessing: Followed the preprocessing steps outlined in the M3DM code

As shown in the attached image, there are significant performance gaps for the dowel, foam, rope, and tire categories. I’m wondering if there might be any details or considerations I may have overlooked that could help improve the results for these categories.
![problem](https://github.com/user-attachments/assets/a3a8564a-0c02-495d-afd7-e63fdbce796a)

Thank you very much for your time and support. Looking forward to your response!

**Comment by LiZhi-CASIA** [comment]
The problem was solved.

**Comment by alex-costanzino** [comment]
Hi, sorry for not getting to you sooner. Would you mind explaining how you solved the problem for future reference?
Many thanks.

**Comment by zhaoyuan1209** [comment]
Can you share the results of your reproduction? My average I-AUROC for reproduction was 94.06. I don't know where I went wrong with the details of my processing.

Metrics for class bagel with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.978   |   0.934   |   0.873  |   0.461  |   0.997 |   0.992
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: cable_gland.: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 108/108 [00:06<00:00, 17.10it/s]
Metrics for class cable_gland with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.963   |   0.891   |   0.801  |   0.398  |   0.989 |   0.865
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: carrot.: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 159/159 [00:25<00:00,  6.14it/s]
Metrics for class carrot with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.982   |   0.947   |   0.894  |   0.489  |   0.998 |   0.983
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: cookie.: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 131/131 [01:05<00:00,  1.99it/s]
Metrics for class cookie with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.941   |   0.891   |   0.833  |   0.463  |   0.971 |   0.994
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: dowel.: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 130/130 [00:45<00:00,  2.84it/s]
Metrics for class dowel with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.944   |   0.836   |   0.740  |   0.377  |   0.985 |   0.983
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: foam.: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:33<00:00,  2.97it/s]
Metrics for class foam with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.962   |   0.890   |   0.805  |   0.404  |   0.992 |   0.895
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: peach.: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 132/132 [00:45<00:00,  2.88it/s]
Metrics for class peach with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.981   |   0.943   |   0.887  |   0.471  |   0.998 |   0.946
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: potato.: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 114/114 [00:35<00:00,  3.23it/s]
Metrics for class potato with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.983   |   0.948   |   0.897  |   0.494  |   0.998 |   0.911
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: rope.: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 101/101 [00:57<00:00,  1.75it/s]
Metrics for class rope with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.974   |   0.924   |   0.862  |   0.462  |   0.998 |   0.965
anaconda3/envs/pytorch2_backup/lib/python3.10/site-packages/pointnet2_ops/pointnet2_utils.py:15: UserWarning: Unable to load pointnet2_ops cpp extension. JIT Compiling.
  warnings.warn("Unable to load pointnet2_ops cpp extension. JIT Compiling.")
Extracting feature from class: tire.: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 112/112 [00:08<00:00, 13.57it/s]
Metrics for class tire with 50ep_4bs
AUPRO@30% | AUPRO@10% | AUPRO@5% | AUPRO@1% | P-AUROC | I-AUROC
  0.979   |   0.937   |   0.879  |   0.469  |   0.996 |   0.872

**Comment by wsjxdy** [comment]
@LiZhi-CASIA Hello, thank you for sharing your valuable experience.
At first, I processed the dataset using the preprocessing code from M3DM, and the results I obtained were similar to yours. Following your advice, I reprocessed the original dataset using the preprocessing method in this project, but the performance is still unsatisfactory, with a large gap compared to the results reported in the paper.
May I ask whether your hyperparameters, random seeds, etc., are consistent with those provided by the authors?
I'm quite frustrated that I haven't been able to reproduce decent performance.
Looking forward to your reply!

<img width="517" height="604" alt="Image" src="https://github.com/user-attachments/assets/0e7b5ee9-6ca0-4c84-b5d0-835068857120" />
